DOCUMENTATION.md

```diff
@@ -43,6 +45,190 @@ The overall territory is laid out here in this [C4](https://c4model.com) System
 
 ![Propulsion c4model.com Container Diagram: Reactors](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.github.com/jet/propulsion/master/diagrams/ReactorsContainer.puml&fmt=svg)
 
+# The `Propulsion.Streams` Programming Model for Projections, Reactions and Workflows
+
+`Propulsion.Streams` provides a programming model to manage running of _Handlers_ in a manner that optimises for the following:
+- isolating the handlers from the client libraries of a given Event Store (the Handler is triggered via a 'Sink' that manages accepts incoming events and checkpointing of progress in a Store-specific manner relevant to the hosting environment in which your handler will be run).
+- providing a clean approach to the testing of Reaction logic with and without involving your actual Event Store (the `MemoryStoreProjector` component is a key part of that story).
+- exposing thorough diagnostic information that enables one to effectively monitor and tune your Handlers in the context of an overall production system.
+
+## Overview of running projections with Propulsion
+
+### Glossary
+
+#### Tranches
+
+A Propulsion source can optionally split it's reading into multiple independent _Tranches_ (a tranche is a portion/division of a pool, it literally means slice in french). Each tranche is read (including reading up to a specified number of batches) and checkpointed independently. The internal Submitter component takes a batch from each Tranche in turn to ensure fair distribution of work across all Tranches that have work to be processed.
+
+### Supported Inputs
```

Comment:
```suggestion
### Supported input events storage engines
```

Reply:
Used "Supported Input Sources" (Feed allows you to wire in arbitrary stuff that has nothing to do with storage engines, and the kafka support allows you to pull stuff that's coming from a storage engine "over a longer wire" (and it is DEFINITELY NOT a storage engine!))